packages/teleport-project-plugin-inline-fetch/src/next/utils.ts

```diff
@@ -74,19 +69,29 @@ export const createNextComponentInlineFetchPlugin: ComponentPluginFactory<Contex
       )
 
       if (extractedResources[propKey]) {
-        const usedResource = resources.items[extractedResources[propKey].id]
-        const resourceName = StringUtils.createStateOrPropStoringValue(
-          usedResource.name + 'Resource'
-        )
-
-        const resourcesPath = paths.resources
-        const currentPagePath = [...paths.pages, ...uidl.outputOptions.folderPath]
+        const extractedResource = extractedResources[propKey]
+        let resourceName: string
+        let dependencyPath: string
 
-        dependencies[resourceName] = {
-          type: 'local',
-          path:
+        if ('id' in extractedResource) {
```

Comment:
This is to make `TS` happy, the regular `extractedResource?.id` doesn't seem to detect the resource as a local resource.